submissions/jeff-james/scripts/article.js

```diff
@@ -52,12 +53,31 @@ Article.fetchAll = function() {
     // Article.loadAll(//TODO: Process our localStorage!
     // Tip: Be careful when handling different data types between localStorage!
     // );
-    console.log('local storage exists');
-    var localData = localStorage.getItem('data');
-    Article.loadAll(JSON.parse(localData));
-    //TODO: Now call the correct method here that will render the index page.
-    // Article.loadAll();
-    articleView.initIndexPage();
+    $.ajax({
+      url: 'data/hackerIpsum.json',
+      success: function (data, message, xhr) {
+        var eTag = xhr.getResponseHeader('eTag');
+        console.log('eTag = ' + eTag);
+        var compareETag = JSON.parse(localStorage.getItem('eTag1'));
+        console.log('eTag1 = ' + compareETag);
+        // console.log(message, eTag);
+        if (eTag !== compareETag) {
+          console.log('are not the same');
+          $.getJSON('data/hackerIpsum.json', function(data){
```

Comment:
But if you don't get the headers, then you already have the data you want and don't have to make another call, like you do here. There's no need for this, you already have data from further up.